selenium_tests/projects/location_relationships.py

```diff
@@ -20,28 +20,17 @@ def test_attach_relationship_to_new_location(self):
         self.wd.find_element_by_link_text(Project.get_test_proj_name()).click()
         self.wd.wait_for_xpath("//h2[contains(text(), 'Project Overview')]")
         self.wd.find_element_by_link_text("Add location").click()
-
-        self.wd.wait_for_xpath("//h3[contains(text(), 'Draw location on map')]")
-        page_state = self.wd.execute_script('return document.readyState;')
-        while page_state != 'complete':
-            page_state = self.wd.execute_script('return document.readyState;')
-
-        self.wd.find_element_by_css_selector("a.leaflet-draw-draw-marker").click()
+        self.wd.wait_for_css("a.leaflet-draw-draw-marker").click()
```

Comment:
The purpose of the lines deleted is to wait until the draw-marker button is visible.
They can be replaced by a one liner.